controllers/reconcilers/kserve_raw_route_reconciler.go

```diff
@@ -0,0 +1,195 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"github.com/opendatahub-io/odh-model-controller/controllers/comparators"
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	"github.com/opendatahub-io/odh-model-controller/controllers/processors"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+	v1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+const (
+	isvcServicePort = 8888
+	authServicePort = 8443
+)
+
+var _ SubResourceReconciler = (*KserveRawRouteReconciler)(nil)
+
+type KserveRawRouteReconciler struct {
+	client         client.Client
+	routeHandler   resources.RouteHandler
+	deltaProcessor processors.DeltaProcessor
+}
+
+func NewKserveRawRouteReconciler(client client.Client) *KserveRawRouteReconciler {
+	return &KserveRawRouteReconciler{
+		client:         client,
+		routeHandler:   resources.NewRouteHandler(client),
+		deltaProcessor: processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveRawRouteReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
```

Comment:
For exposing, KServe supports creating `Ingress` resources, but we have it [turned off](https://github.com/opendatahub-io/kserve/blob/ae30b971320f45bc98dd362b6b8873d7c1a48873/config/overlays/odh/inferenceservice-config-patch.yaml#L28). [OpenShift can follow-up](https://docs.openshift.com/container-platform/4.16/networking/routes/route-configuration.html#nw-ingress-creating-a-route-via-an-ingress_route-configuration) and create Routes.

Given the push for better supporting the API, rather than doing our own thing, we should try to align better with upstream.

By creating Routes on our side, KServe is no longer aware. So, it won't expose what's the endpoint in `status.url` of the InferenceService. This will just increase complexity on dashboard code. Also, it will be more complex for users because the `status.url` field will be useless for Raw, but will be useful for Serverless.

Reply:
Ah nice, I realized the thing about the `status.url` myself but didn't realize an option existed to do this in a way such that Kserve is aware. I'll incorporate this.